libs/backend/src/scan/cast_vote_records/export.ts

```diff
@@ -0,0 +1,292 @@
+        : [sideOneFilename, sideTwoFilename];
+
+      yield buildCastVoteRecord({
+        election,
+        electionId,
+        scannerId,
+        castVoteRecordId: unsafeParse(BallotIdSchema, id),
+        batchId,
+        ballotMarkingMode: 'hand',
+        definiteMarkThreshold,
+        pages: [
+          {
+            interpretation: front,
+            inlineBallotImage: hasWriteIns(front)
+              ? await getWriteInImage(frontFilename)
+              : undefined,
+          },
+          {
+            interpretation: back,
+            inlineBallotImage: hasWriteIns(back)
+              ? await getWriteInImage(backFilename)
+              : undefined,
+          },
+        ],
+        ballotPageLayoutsLookup,
+      });
+    }
+  }
+}
+
+async function* concatStreams(readables: NodeJS.ReadableStream[]) {
+  for (const readable of readables) {
+    for await (const chunk of readable) {
+      yield chunk;
+    }
+  }
+}
+
+interface GetCastVoteRecordReportStreamParams
+  extends GetCastVoteRecordGeneratorParams {
+  isTestMode: boolean;
+  batchInfo: BatchInfo[];
+}
+
+function getCastVoteRecordReportStream({
+  election,
+  electionId,
+  scannerId,
+  definiteMarkThreshold,
+  isTestMode,
+  ballotPageLayoutsLookup,
+  resultSheetGenerator,
+  batchInfo,
+}: GetCastVoteRecordReportStreamParams): NodeJS.ReadableStream {
+  const castVoteRecordGenerator = getCastVoteRecordGenerator({
+    election,
+    electionId,
+    scannerId,
+    definiteMarkThreshold,
+    ballotPageLayoutsLookup,
+    resultSheetGenerator,
+  });
+
+  const reportMetadata = buildCastVoteRecordReport({
+    election,
+    electionId,
+    generatingDeviceId: scannerId,
+    scannerIds: [scannerId],
+    reportTypes: [CVR.ReportType.OriginatingDeviceExport],
+    isTestMode,
+    batchInfo,
+  });
+
+  const reportMetadataJson = JSON.stringify(reportMetadata, undefined, 2);
+
+  const reportMetadataStream = Readable.from(
+    reportMetadataJson.replace(/\n\}$/, ',\n  "CVR": [\n')
+  );
+
+  const cvrStream = Readable.from(
+    mapAsync(
+      castVoteRecordGenerator,
+      (cvr) => `    ${JSON.stringify(cvr, undefined, 2)},\n`
+    )
+  );
+
+  const closingStream = Readable.from(`  ]\n}`);
+
+  return Readable.from(
+    concatStreams([reportMetadataStream, cvrStream, closingStream])
+  );
+}
+
+interface ExportCastVoteRecordReportToUsbDriveParams {
+  electionHash: string;
+  ballotsCounted: number;
+  election: Election;
+  definiteMarkThreshold: number;
+  ballotPageLayoutsLookup: BallotPageLayoutsLookup;
+  resultSheetGenerator: Generator<ResultSheet>;
+  isTestMode: boolean;
+  batchInfo: BatchInfo[];
+}
+
+/**
+ * Exports a CDF cast vote record report to an inserted and mounted USB drive
+ */
+export async function exportCastVoteRecordReportToUsbDrive({
+  election,
+  electionHash,
+  isTestMode,
+  ballotsCounted,
+  ...rest
+}: ExportCastVoteRecordReportToUsbDriveParams): Promise<
+  Result<void, ExportDataError>
+> {
```

Comment:
The idea was to extract the entire exporting to USB drive into this lib, so the scanners just have to gather the data from their stores and make a call.